macaw-aarch32/src/Data/Macaw/ARM/Identify.hs

```diff
@@ -69,6 +69,13 @@ isValidReturnAddress val0 =
       MC.BVAnd _ val5 _ <- MC.valueAsApp val4
       MC.BVShr _ val6@(MC.RelocatableValue {}) _ <- MC.valueAsApp val5
       return val6
+      <|> do
+      MC.BVOr _ val1 _ <- MC.valueAsApp val0
+      MC.BVShl _ val2 _ <- MC.valueAsApp val1
+      MC.UExt val3 _ <- MC.valueAsApp val2
+      MC.Trunc val4 _ <- MC.valueAsApp val3
+      MC.BVShr _ val5@(MC.RelocatableValue {}) _ <- MC.valueAsApp val4
+      return val5
```

Comment:
The indentation here makes this somewhat more difficult to read. Better, I think, would be to write it like this:

```hs
  _ -> class1 <|> class2
  where
    class1 = do
      MC.BVOr _ val1 _ <- MC.valueAsApp val0
      MC.BVShl _ val2 _ <- MC.valueAsApp val1
      MC.UExt val3 _ <- MC.valueAsApp val2
      MC.Trunc val4 _ <- MC.valueAsApp val3
      MC.BVAnd _ val5 _ <- MC.valueAsApp val4
      MC.BVShr _ val6@(MC.RelocatableValue {}) _ <- MC.valueAsApp val5
      return val6

    class2 = do
      MC.BVOr _ val1 _ <- MC.valueAsApp val0
      MC.BVShl _ val2 _ <- MC.valueAsApp val1
      MC.UExt val3 _ <- MC.valueAsApp val2
      MC.Trunc val4 _ <- MC.valueAsApp val3
      MC.BVShr _ val5@(MC.RelocatableValue {}) _ <- MC.valueAsApp val4
      return val5
```